sql/table.go

```diff
@@ -164,6 +164,8 @@ func encodeTableKey(b []byte, v parser.Datum) ([]byte, error) {
 		return encoding.EncodeNumericFloat(b, float64(t)), nil
 	case parser.DString:
 		return encoding.EncodeBytes(b, []byte(t)), nil
+	case parser.DNull:
+		return b, nil
```

Comment:
Except for PKey-natural ordering (which can't be NULL anyway) we'll always be sorting in memory; since we do need to support low and high `NULL`s as you say, we'll need some special sorting logic (i.e. we can't use the natural ordering of the encoded values directly). I can't see where `NULL`-first encoding buys anything.

This just leaves indexes, which also don't seem to benefit from `NULL`-first encoding.

Does `SQLite` support both high and low `NULL`s? If not, that would be the reason they want the `NULL`-first property.